### PR TITLE
Avoid using a closure in the render loop when pushing model draw commands

### PR DESCRIPTION
**Pull request by ptrgags**
`ModelSceneGraph.pushDrawCommands()` was using an anonymous function, which means a closure was being allocated every frame. with a scratch variable and a private function, I was able to avoid the allocations.

I still want to verify in the DevTools profiler that this makes a noticeable difference, but otherwise this is ready for review @j9liu

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Thanks for the pull request @ptrgags!
* :heavy_check_mark: Signed CLA found.
* :grey_question: [CHANGES.md](https://github.com/CesiumGS/cesium/blob/no-closure/CHANGES.md) was not updated.
    * If this change updates the public API in any way, please add a bullet point to `CHANGES.md`.
* :grey_question: Unit tests were not updated.
    * Make sure you've [updated tests](https://github.com/CesiumGS/cesium/tree/main/Documentation/Contributors/TestingGuide) to reflect your changes, added tests for any new code, and ran the code coverage tool.

Reviewers, don't forget to make sure that:

- [ ] Cesium Viewer works.
- [ ] Works in 2D/CV.

**Comment by ptrgags** [comment]
@j9liu There were a few few other similar closures in `ModelSceneGraph` so I gave them the same treatment. They aren't always called every frame like `pushDrawCommands()` is, but it doesn't hurt to have.

File: Source/Scene/Model/ModelSceneGraph.js [hunk 1]
```diff
@@ -708,7 +708,8 @@ function traverseSceneGraph(
   sceneGraph,
   runtimeNode,
   visibleNodesOnly,
-  callback
+  callback,
+  callbackOptions
```
Comment (j9liu) [inline Source/Scene/Model/ModelSceneGraph.js:712]: Can we add this parameter to the documentation above this function?

File: Source/Scene/Model/ModelSceneGraph.js [hunk 2]
```diff
@@ -803,31 +858,45 @@ ModelSceneGraph.prototype.pushDrawCommands = function (frameState) {
   // Otherwise, the silhouette may draw on top of the model. This requires
   // gathering the original commands and the silhouette commands separately.
```
Comment (j9liu) [inline Source/Scene/Model/ModelSceneGraph.js:860]: nitpick but can we delete the space here?

**Review by j9liu — COMMENTED** [review]
@ptrgags just a few comments. I tried to profile with the Chrome Performance tab using [this Sandcastle](http://localhost:8080/Apps/Sandcastle/index.html#c=bVNbc6IwFP4rGZ90xoYgINDaznZ11sVWqNW2Ow4vEWKbGhI3Cba40/++QbqzvT1wOeG75HyHWBYYS8w1GBJFy2IyBzjLiFJAC1CJUgIqOMBKEa1S3mBgJDjMyRqXTJ8fwAuxIRycgrRFqsnDapzRhE6im31kxzRSEb/2smHUjzbbX7fDSQgN6Hc+3tQgmoziYvo428eVXcTjyL1czHrLRaSTuwjFc1QtRzf76X5axXfT53g/o5fDyXZpxJL5E8V3P1D0KJ7j0f1zMoq8ZHSD1jPorotbvVFLe3i+f7z6vpUT1y+j8gLt5FEwzMrRxdSfBhgVP9y0dZLylGeCKw12lDwRadrg5Ok1Dnh7WGunrexQDwXXmHIi01YX/Ek5APdMrMgxWGOmSMpfOm/0NGXE5GYEG2WoMsIJ3EpaUE13REGc5+1a5I1h83BGi4bcPpgAUEp2DP7Hf02UmU1G4FqK4ryeTpS3+6EToE63Jrx0Ut5s5Z11hgsiMTTwuq9GOydKU461mfPxu41gqc0b5k77yLUd5PcQdB3Xd3uBKbqgF4R92/eg3Q/tAPl+0AVHThAiO/RgGDpe3zW3ZjdCUmJy+2Txk+Cc8vsrqrOHa8FYuw+NuB14xsD2Q893jCaCHkIo8JDt2Ah5pscu+IAL+rWPyb7VbR0mCsC/C4CB0hUjZ00BwDdabIXUdaBtCC1Nii3DplFrVWYbomGmVOfklWm9pQ5yugM0P/3iVwAZMyfEfFmXjM3pnqSts4Fl8J+oTBw6TnZEMlzVsAf77LJZhBAOLFN+zdRCsBWWH5T/Ag). I waited until the window was a black screen, then recorded for 15 seconds, then observed the results.

With closure:
with closure

Without closure:
without closure

So this change does produce a difference, even if relatively small. I'm sure that difference will scale with more models.

**Comment by ptrgags** [comment]
@j9liu updated!

**Comment by j9liu** [comment]
@ptrgags merging now, thanks!

**Pull request merged**